CanvasPlusPlayground/Views/IndividualCourseListView.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  IndividualCourseListView.swift
+//  CanvasPlusPlayground
+//
+//  Created by Vamsi Putti on 10/3/24.
+//
+
+import SwiftUI
+
+struct IndividualCourseListView: View {
+    @Environment(CourseManager.self) var courseManager
+    @State var isPref = false
+    let course: Course
+    
+    var body: some View {
+        @Bindable var courseManager = courseManager
+        
+        HStack {
+            Button {
+                isPref.toggle()
```

Comment:
I think you can migrate this logic to `CourseManager`.

```swift
Button {
  courseManager.toggleCoursePref(course: course)
} label: {
  // …
}
```

And then you wouldn’t need the `isPref` bool either here. You can create a function in `CourseManager` for that.